Services/LTI/classes/Screen/LtiViewLayoutProvider.php

```diff
@@ -18,75 +29,173 @@
 class LtiViewLayoutProvider extends AbstractModificationProvider implements ModificationProvider
 {
 
+    const GS_EXIT_LTI = 'lti_exit_mode';
+
+    protected function isLTIMode(): bool
+    {
+        return $this->dic["lti"]->isActive();
+    }
+
+    protected function isLTIExitMode(CalledContexts $screen_context_stack): bool
+    {
+        $data_collection = $screen_context_stack->current()->getAdditionalData();
+        return $data_collection->is(self::GS_EXIT_LTI, true);
+    }
+
+    public function isInterestedInContexts() : ContextCollection
+    {
+        return $this->context_collection->lti();
+    }
+
     /**
      * @inheritDoc
      */
-     
-    public function isInterestedInContexts() : ContextCollection
+    public function getPageBuilderDecorator(CalledContexts $screen_context_stack) : ?PageBuilderModification
     {
-        return $this->context_collection->lti();
+        if(! $this->isLTIMode()) {
+            return null;
+        }
+
+        $this->globalScreen()->layout()->meta()->addCss('./Services/LTI/templates/default/lti.css');
+        if (isset($_SESSION['lti_launch_css_url']) && $_SESSION['lti_launch_css_url'] != "") {
+            $this->globalScreen()->layout()->meta()->addCss($_SESSION['lti_launch_css_url']);
+        }
+
+        return $this->factory->page()
+            ->withModification(
+                function (PagePartProvider $parts): Page {
+                    $p = new StandardPageBuilder();
```

Comment:
I just realized I should pass the current PageBuilder as well, fine for me at the moment, maybe I change that later.